nucypher/datastore/ritual.py

```diff
@@ -1,23 +1,81 @@
+import abc
+from abc import abstractmethod
 from collections import defaultdict
 from typing import List, Optional
 
 from atxm.tx import AsyncTx
-from nucypher_core.ferveo import (
-    Validator,
-)
+from nucypher_core.ferveo import Validator
 
 from nucypher.blockchain.eth.models import (
+    DKG_PHASE_1,
+    DKG_PHASE_2,
     HANDOVER_AWAITING_BLINDED_SHARE,
     HANDOVER_AWAITING_TRANSCRIPT,
-    PHASE1,
-    PHASE2,
+    SIGNING_AWAITING_SIGNATURES,
     Coordinator,
 )
-from nucypher.types import PhaseId
+from nucypher.types import PhaseId, PhaseNumber
+
+
+class RitualStorage(abc.ABC):
+    """A simple in-memory storage for ritual data"""
+
+    _KEYS: List[str] = NotImplemented
+
+    def __init__(self):
+        self._data = defaultdict(dict)
+
+    def clear(self, ritual_id):
+        for key in self._KEYS:
+            try:
+                del self._data[key][ritual_id]
+            except KeyError:
+                continue
+
+    #
+    # DKG Phases
+    #
+    @classmethod
+    @abstractmethod
+    def _get_phase_key(cls, phase: PhaseNumber):
+        raise NotImplementedError
+
+    def store_ritual_phase_async_tx(self, phase_id: PhaseId, async_tx: AsyncTx):
+        key = self._get_phase_key(phase_id.phase)
+        self._data[key][phase_id.ritual_id] = async_tx
```

Comment:
Just thinking aloud. We could do this so we are reusing this internal function, right? Maybe this is better in terms of code maintainability.

```suggestion
    def store_ritual_phase_async_tx(self, phase_id: PhaseId, async_tx: AsyncTx):
        key = self._get_phase_key(phase_id.phase)
        self._store_ritual_metadata(key, phase_id.ritual_id)
```

Reply:
✔️ - great catch for all of the suggestions. Fixed via https://github.com/nucypher/nucypher/pull/3645/commits/4c93abe71ebf55cbafe6f4ae40675a02813b1d88